### PR TITLE
20635: Add max bgp hop limit for non-rfc1918 cidrs

### PR DESCRIPTION
**Pull request by mlin-aviatrix**

### CODE REVIEW

File: aviatrix/resource_aviatrix_controller_bgp_max_as_limit_config.go [hunk 1]
```diff
@@ -69,10 +91,34 @@ func resourceAviatrixControllerBgpMaxAsLimitConfigUpdate(ctx context.Context, d
 	client := meta.(*goaviatrix.Client)
```
Comment (CyrusJavan) [inline aviatrix/resource_aviatrix_controller_bgp_max_as_limit_config.go:91]: We should do this same validation in update:
```go
if !maxAsLimitOk && !maxAslimitNonRfc1918Ok {
		return diag.Errorf("error updating controller BGP max AS limit config: at least one of max_as_limit and max_as_limit_non_rfc1918 must be provided")
	}
```

File: aviatrix/resource_aviatrix_controller_bgp_max_as_limit_config.go [hunk 2]
```diff
@@ -69,10 +91,34 @@ func resourceAviatrixControllerBgpMaxAsLimitConfigUpdate(ctx context.Context, d
...
+			maxAsLimit := d.Get("max_as_limit").(int)
+			err := client.SetControllerBgpMaxAsLimit(ctx, maxAsLimit)
+			if err != nil {
+				return diag.Errorf("failed to update controller BGP max AS limit config: %v", err)
+			}
+		} else {
+			err := client.DisableControllerBgpMaxAsLimit(ctx)
+			if err != nil {
+				return diag.Errorf("failed to update controller BGP max AS limit config: %v", err)
+			}
```
Comment (CyrusJavan) [inline aviatrix/resource_aviatrix_controller_bgp_max_as_limit_config.go]: These error strings need to be unique, same below

File: aviatrix/resource_aviatrix_controller_bgp_max_as_limit_config.go [hunk 3]
```diff
@@ -69,10 +91,34 @@ func resourceAviatrixControllerBgpMaxAsLimitConfigUpdate(ctx context.Context, d
 	client := meta.(*goaviatrix.Client)
 
 	if d.HasChange("max_as_limit") {
-		maxAsLimit := d.Get("max_as_limit").(int)
-		err := client.SetControllerBgpMaxAsLimit(ctx, maxAsLimit)
-		if err != nil {
-			return diag.Errorf("failed to update controller BGP max AS limit config: %v", err)
+		_, maxAsLimitOk := d.GetOk("max_as_limit")
+		if maxAsLimitOk {
+			maxAsLimit := d.Get("max_as_limit").(int)
```
Comment (CyrusJavan) [inline aviatrix/resource_aviatrix_controller_bgp_max_as_limit_config.go]: Why not just get the maxAsLimit when you call GetOk above?

File: aviatrix/resource_aviatrix_controller_bgp_max_as_limit_config_test.go [hunk 4]
```diff
@@ -46,6 +47,7 @@ func testAccControllerBgpMaxAsLimitConfigBasic() string {
 	return `
 resource "aviatrix_controller_bgp_max_as_limit_config" "test_bgp_max_as_limit" {
 	max_as_limit = 1
+	max_as_limit_non_rfc1918 = 2
```
Comment (CyrusJavan) [inline aviatrix/resource_aviatrix_controller_bgp_max_as_limit_config_test.go]: formatting

File: docs/resources/aviatrix_controller_bgp_max_as_limit_config.md [hunk 5]
```diff
@@ -16,6 +16,7 @@ The **aviatrix_controller_bgp_max_as_limit_config** resource allows management o
 # Create an Aviatrix Controller BGP max AS limit config
 resource "aviatrix_controller_bgp_max_as_limit_config" "test_max_as_limit" {
   max_as_limit = 1
+  max_as_limit_non_rfc1918 = 2
```
Comment (CyrusJavan) [inline docs/resources/aviatrix_controller_bgp_max_as_limit_config.md]: formatting

File: goaviatrix/controller_bgp_max_as_limit_config.go [hunk 6]
```diff
@@ -39,31 +55,65 @@ func (c *Client) DisableControllerBgpMaxAsLimit(ctx context.Context) error {
...
+	if resp.Results.MaxAsLimit == "" && resp.Results.MaxAsLimitNonRfc1918 == "" {
+		return 0, 0, ErrNotFound
 	}
 
-	maxAsLimit, err := strconv.Atoi(resp.Results)
-	if err != nil {
-		return 0, fmt.Errorf("error converting max_as_limit to int: %v", err)
+	var maxAsLimit, maxAsLimitNonRfc1918 int
+	if resp.Results.MaxAsLimit == "" {
+		maxAsLimit = 0
```
Comment (CyrusJavan) [inline goaviatrix/controller_bgp_max_as_limit_config.go]: not needed, in Go values are initialized to their zero value, maxAsLimit is already `0`

**Review by CyrusJavan — CHANGES_REQUESTED** [review]

File: aviatrix/resource_aviatrix_controller_bgp_max_as_limit_config_test.go (hunk 4 above)
Comment (mlin-aviatrix) [reply aviatrix/resource_aviatrix_controller_bgp_max_as_limit_config_test.go]: What needs to be changed here?

Comment (CyrusJavan) [reply aviatrix/resource_aviatrix_controller_bgp_max_as_limit_config_test.go]: Just need to line up the `=`

**Review by CyrusJavan — APPROVED** [review]
Good job!

**Review by zhiqiangjin — APPROVED** [review]

**Pull request merged**